### PR TITLE
Add docker pull end to end test

### PR DESCRIPTION
**Pull request by mirzal**
Adding this, as downstream [bz#1661019](https://bugzilla.redhat.com/show_bug.cgi?id=1661019) has finally been resolved and this case has a chance to pass.

This is very similar to already existing `test_positive_container_admin_end_to_end_search`. As name suggest, this one is for `docker pull`, the other one is for `docker search`.

It might seem similar to `test_positive_pull_image`. Difference is, `positive_pull_image` tries to pull image while talking with crane directly. This one talks to Katello, which is responsible for processing and forwarding requests to crane under the hood. It also covers Katello-specific functionality.

```
$ pytest -v tests/foreman/cli/test_docker.py::DockerClientTestCase::test_positive_container_admin_end_to_end_pull

tests/foreman/cli/test_docker.py::DockerClientTestCase::test_positive_container_admin_end_to_end_pull PASSED                                                                            [100%]
```

### CODE REVIEW

File: tests/foreman/cli/test_docker.py [hunk 1]
```diff
@@ -1815,6 +1815,137 @@ def test_positive_container_admin_end_to_end_search(self):
...
+            'id': content_view['versions'][0]['id'],
+            'to-lifecycle-environment-id': lce['id'],
+        })
+        LifecycleEnvironment.update({
+            'registry-name-pattern': registry_name_pattern,
+            'registry-unauthenticated-pull': 'false',
+            'id': lce['id'],
+            'organization-id': self.org['id'],
+        })
+        docker_repo_uri = " {}/{}-{}/{} ".format(
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_docker.py]: " {}/{}-{}/{} " There was space in start and end, was that expected?

File: tests/foreman/cli/test_docker.py [hunk 2]
```diff
@@ -1815,6 +1815,137 @@ def test_positive_container_admin_end_to_end_search(self):
...
+                settings.server.hostname,
+                pattern_prefix,
+                content_view['label'],
+                docker_upstream_name
+        ).lower()
+
+        # 3. Try to pull in docker image from Satellite
+        remote_pull_command = 'docker pull {0}'.format(docker_repo_uri)
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_docker.py]: 'remote_pull_command'  can we rephrase to just 'docker_pull_command', what do you think?

File: tests/foreman/cli/test_docker.py [hunk 3]
```diff
@@ -1815,6 +1815,137 @@ def test_positive_container_admin_end_to_end_search(self):
...
+        # publishing takes few seconds sometimes
+        retries = 20 #if bz_bug_is_open(1452149) else 1
+        for i in range(retries):
+            result = ssh.command(
+                    remote_pull_command,
+                    hostname=self.docker_host.ip_addr
+            )
+            if result.return_code == 0:
+                break
+            sleep(2)
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_docker.py]: Could have used the wait_for similar to https://github.com/omkarkhatavkar/robottelo/blob/892f122a1258d27c740f088eb8f4ce4d1d597134/tests/foreman/cli/test_repository.py#L134

File: tests/foreman/cli/test_docker.py (hunk 2 above)
Comment (mirzal) [reply tests/foreman/cli/test_docker.py]: Changed.

File: tests/foreman/cli/test_docker.py (hunk 1 above)
Comment (mirzal) [reply tests/foreman/cli/test_docker.py]: Good catch!
As far as I can tell, these spaces didn't have any impact on functionality, but they aren't necessary either.

File: tests/foreman/cli/test_docker.py (hunk 3 above)
Comment (mirzal) [reply tests/foreman/cli/test_docker.py]: We can, although we don't use wait_for in this particular file, and another case there does use the same idiom with sleep.

Let's leave it as is and switch both these tests to `wait_for` in separate PR. How does it sound?

Comment (omkarkhatavkar) [reply tests/foreman/cli/test_docker.py]: Ok

Comment (oshtaier) [reply tests/foreman/cli/test_docker.py]: On other hand, we can do that in scope of current PR =) Static sleeps and loops like this are bad style. We are trying to follow only good practice in the code, so if have issue in it, it doesn't make sense to submit bad code and then change it in all places

**Comment by oshtaier** [comment]
@omkarkhatavkar why do you resolve comment?
@mirzal Why do you merge PR by yourself?

**Comment by omkarkhatavkar** [comment]
@oshtaier : I did not saw your comment before I resolve it. If you have commented little early I would not have resolved that.

**Comment by oshtaier** [comment]
@omkarkhatavkar okay

**Comment by oshtaier** [comment]
@mirzal are you going to create PR to fix issue in both places or should we revert that PR?

**Comment by mirzal** [comment]
@oshtaier I have merged it because all formal requirements for merge were already met.

Yes, I will create another PR to switch sleep to `wait_for`, once I have more pressing matters resolved. We had `sleep` inside `test_positive_pull_image` case for over 1,5 year, we can have it for day or two more.

**Comment by oshtaier** [comment]
Okay, first of all - you merged PR by yourself - which is direct rule violation, we never do like that (or only in case we have really bad day which is not a good enough reason =)). Second - you ignored my comment (and that happen when you ignore rule number one - never merge your own code). Of course, bad code can wait and so can wait your test coverage. Even if we have `sleep` there for twenty years - that means nothing, and especially that we need to populate more code like this.
I don't want to escalate that more, so let's just leave things like that, but for the future - that is really bad tone of cooperation

**Comment by mirzal** [comment]
@oshtaier In https://github.com/SatelliteQE/robottelo/pull/6728#pullrequestreview-201854985 Omkar raised a point about using `wait_for` instead of sleep. I agreed and proposed solution on how we can proceed with it. He agreed. Then you added your opinion. I appreciate it, but as far as I can tell, it was fundamentally repeating Omkar's point, which we already resolved.

If I have misunderstood and it was about something else, it might have helped to raise it in separate thread. If you believe it was important enough to prevent merge, you could have maximize visibility by also giving NACK.

I agree, we should all strive for effective collaboration. We can achieve it only when we work on reaching mutual agreements and accept that other people might do things differently than we would do them.

Followup PR is #6733

**Comment by oshtaier** [comment]
@mirzal sure, it was not blocking comment, so no reason for NACK. My only concern that you make decision (and following merge) based only on your opinion. That is essential reason why reviewers only can merge the code. So please don't do it.
Once more, concern not in the PR code, but on violation of the rules and your decisions in the places where they should not appear in the first place.

I would be happy to close that discussion. Thanks for your understanding.

**Pull request merged**